lib/logstash/plugin_mixins/jdbc/statement_handler.rb

```diff
@@ -16,27 +16,39 @@ def self.build_statement_handler(plugin, logger)
           klass = NormalStatementHandler
         end
       end
-      klass.new(plugin, logger)
+      klass.new(plugin)
     end
 
-    attr_reader :statement, :parameters, :statement_logger
+    attr_reader :statement, :parameters
 
-    def initialize(plugin, statement_logger)
+    def initialize(plugin)
       @statement = plugin.statement
-      @statement_logger = statement_logger
-      post_init(plugin)
     end
 
     def build_query(db, sql_last_value)
-      # override in subclass
+      fail NotImplementedError # override in subclass
     end
 
-    def post_init(plugin)
-      # override in subclass, if needed
-    end
   end
 
   class NormalStatementHandler < StatementHandler
+
+    attr_reader :parameters
+
+    def initialize(plugin)
+      super(plugin)
+      @parameter_keys = ["sql_last_value"] + plugin.parameters.keys
+      @parameters = plugin.parameters.inject({}) do |hash,(k,v)|
```

Comment:
This loop modify the `parameters` converting `LogStash::Timestamp` parameters to something else (Joda Time in this case). Maybe extracting in a method would make the code more explicit.

Reply:
yeah this is confusing but it pre-existed - just a refactoring of a `post_init` method: https://github.com/logstash-plugins/logstash-integration-jdbc/pull/109/files#diff-18a21e2901a3bdc0983fab9b9c06cf6575f609433317586a8368e815009777e7L62-L70